tensorflow/core/kernels/mkl_quantize_op.cc

```diff
@@ -57,24 +244,100 @@ class MklQuantizeV2Op : public OpKernel {
+    if (round_mode_string == "HALF_TO_EVEN") {
+      OP_REQUIRES(ctx, mode_string == "SCALED",
+                  errors::InvalidArgument("Round mode 'HALF_TO_EVEN' "
+                                          "only supported for mode 'SCALED', "
+                                          "but mode is '" +
+                                          mode_string + "'."));
+      round_mode_ = ROUND_HALF_TO_EVEN;
+    }
   }
 
-  ~MklQuantizeV2Op() {}
+  ~MklQuantizeV2Op() {
+    if (minfirst_input_ != nullptr) {
+      delete minfirst_input_;
+      minfirst_input_ = nullptr;
+    }
+  }
+
+  float* GetMinfirstInputBuf(int size) {
+    if (!minfirst_input_) {
+      minfirst_input_ = new float[size];
+      minfirst_input_size_ = size;
+    } else if (size != minfirst_input_size_) {
+      delete minfirst_input_;
+      minfirst_input_ = new float[size];
+      minfirst_input_size_ = size;
+    }
+
+    return minfirst_input_;
+  }
+
+  void Compute_Scalar(OpKernelContext* ctx, float min_range, float max_range) {
+    // TO-DO - Scalar support has to be added for SCALE mode
+    OP_REQUIRES(ctx, (mode_ == QUANTIZE_MODE_MIN_FIRST),
+                errors::InvalidArgument(
+                    "Scalar calculation in MKL is supported only for"
+                    "MIN_FIRST mode for now."));
+
+    auto cpu_engine = engine(engine::cpu, 0);
+    const Tensor& input = ctx->input(0);
+    const unsigned int src_idx = 0;
+    const Tensor& src_tensor = MklGetInput(ctx, src_idx);
+
+    MklDnnShape output_mkl_shape;
+    output_mkl_shape.SetMklTensor(false);
+
+    Tensor* output_tensor = nullptr;
+    AllocateOutputSetMklShape(ctx, 0, &output_tensor, src_tensor.shape(),
+                              output_mkl_shape);
+    TensorShape min_tf_shape = {};
+    MklDnnShape min_mkl_shape;
+    min_mkl_shape.SetMklTensor(false);
+    Tensor* output_min_tensor = nullptr;
+    AllocateOutputSetMklShape(ctx, 1, &output_min_tensor, min_tf_shape,
+                              min_mkl_shape);
+    TensorShape max_tf_shape = {};
+    MklDnnShape max_mkl_shape;
+    max_mkl_shape.SetMklTensor(false);
+    Tensor* output_max_tensor = nullptr;
+    AllocateOutputSetMklShape(ctx, 2, &output_max_tensor, max_tf_shape,
+                              max_mkl_shape);
+
+    // Estimate scale for qunatization
+    float scale_factor = 0;
+    const int number_of_bits = sizeof(T) * 8;
+    const int64 number_of_steps = static_cast<int64>(1) << number_of_bits;
+    scale_factor = (number_of_steps - 1.0) / (max_range - min_range);
+
+    float* src_data = const_cast<float*>(src_tensor.flat<float>().data());
+    T* out_data = output_tensor->flat<T>().data();
+
+    out_data[0] = (src_data[0] - min_range) * scale_factor;
+    output_min_tensor->flat<float>()(0) = min_range;
+    output_max_tensor->flat<float>()(0) = max_range;
+
+    return;
+  }
 
   void Compute(OpKernelContext* ctx) override {
+    const Tensor& input = ctx->input(0);
```

Comment:
Maybe move `src_index` declaration to above this line and use it?
```suggestion
    const Tensor& input = ctx->input(src_idx);
```

Reply:
Done